src/mongo/mongo.ts

```diff
@@ -440,12 +434,9 @@ export class Collection implements IMongoResource {
 		return this.db.dropCollection(this.collection.collectionName);
 	}
 
-	private find(args?: any): Promise<string> {
-		const promise = new Promise((c, e) => {
-			let cursor = this.collection.find(args);
-			this.readNext([], cursor, 20, c);
-		});
-		return promise;
+	private find(args?: any): Thenable<string> {
+		return this.collection.find(args).limit(20)
```

Comment:
I was just maintaining the previous behavior. I changed it to be a const and filed an issue to improve this behavior: https://github.com/Microsoft/vscode-mongodb/issues/31